src/backend/distributed/transaction/transaction_management.c

```diff
@@ -256,6 +255,8 @@ CoordinatedTransactionCallback(XactEvent event, void *arg)
 
 		case XACT_EVENT_PRE_COMMIT:
 		{
+			RemoveIntermediateResultsDirectory();
+
```

Comment:
We can probably move this below the if statement below to not bother non-citus transactions.

Reply:
Agreed; the function is basically a no-op if not in a Citus transaction but it's such a simple change to not call it needlessly I think it makes sense to move.